### PR TITLE
Update RegEx to detect hyphenated test names

### PR DESCRIPTION
**Pull request by gauravpathak**
openqa-clone-job was ignoring test names with hyphens in it such as `WORKER_CLASS:ovs-server+=,openqaworker27`.
Modify the regular expression in clone_job_apply_settings function to detect and capture test names with any character

### CODE REVIEW

**Comment by perlpunk** [comment]
https://app.circleci.com/pipelines/github/os-autoinst/openQA/15269/workflows/e3618cfd-f27f-411b-a7fa-05f86fb0f988/jobs/146112
```
[07:14:15] t/24-worker-engine.t ...................................... 1/? 
#   Failed test 'no (unexpected) warnings (via done_testing)'
#   at t/24-worker-engine.t line 549.
# Got the following unexpected warnings:
#   1: Use of uninitialized value $string in substitution (s///) at /usr/lib/perl5/vendor_perl/5.26.1/Mojo/IOLoop/ReadWriteProcess/CGroup.pm line 26.
#   2: Use of uninitialized value $string in substr at /usr/lib/perl5/vendor_perl/5.26.1/Mojo/IOLoop/ReadWriteProcess/CGroup.pm line 28.
# Looks like you failed 1 test of 13.
[07:14:15] t/24-worker-engine.t ...................................... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/13 subtests 
```
I've seen this error elsewhere recently. Ill create a ticket and retrigger the test.

File: lib/OpenQA/Script/CloneJob.pm [hunk 1]
```diff
@@ -39,7 +39,7 @@ sub clone_job_apply_settings ($argv, $depth, $settings, $options) {
 
     for my $arg (@$argv) {
         # split arg into key and value
-        unless ($arg =~ /([A-Z0-9_]+)(:(\w+))?(\+)?=(.*)/) {
+        unless ($arg =~ /([A-Z0-9_]+)(:(\w.+?))?(\+)?=(.*)/) {
```
Comment (perlpunk) [inline lib/OpenQA/Script/CloneJob.pm:42]: I'm wondering if we should really allow any character. I actually don't know if we have any restrictions on test names, would be good to check.
```suggestion
        unless ($arg =~ /([A-Z0-9_]+)(:([\w-]+?))?(\+)?=(.*)/) {
```

Comment (gauravpathak) [reply lib/OpenQA/Script/CloneJob.pm:42]: let me try  the suggested change.

**Review by Martchus — APPROVED** [review]

**Comment by perlpunk** [comment]
Please squash your commits to one

**Review by perlpunk — APPROVED** [review]

**Pull request merged**